internal/generate/olm-catalog/package_manifest.go

```diff
@@ -60,21 +63,41 @@ func NewPackageManifest(cfg gen.Config, csvVersion, channel string, isDefault bo
 		channelIsDefault: isDefault,
 		fileName:         getPkgFileName(cfg.OperatorName),
 	}
-	if g.Inputs == nil {
-		g.Inputs = map[string]string{}
-	}
-	if manifests, ok := g.Inputs[ManifestsDirKey]; !ok || manifests == "" {
-		g.Inputs[ManifestsDirKey] = filepath.Join(OLMCatalogDir, g.OperatorName)
-	}
+
+	// Pkg manifest generator has no defined inputs
+	g.Inputs = map[string]string{}
+
+	// The olm-catalog directory location depends on where the output directory is set.
 	if g.OutputDir == "" {
-		g.OutputDir = filepath.Join(OLMCatalogDir, g.OperatorName)
+		g.OutputDir = scaffold.DeployDir
+	}
+	olmCatalogDir := filepath.Join(g.OutputDir, OLMCatalogChildDir)
+
+	// Check if the generator should update from an existing package manifest file
+	pkgManifestDirPath := filepath.Join(olmCatalogDir, g.OperatorName)
+	pkgManifestFilePath := filepath.Join(pkgManifestDirPath, g.fileName)
+	if isFileExist(pkgManifestFilePath) {
+		g.existingPkgManifestDir = pkgManifestDirPath
 	}
+
 	return g
 }
 
+func isFileExist(path string) bool {
```

Comment:
Similar comment to that for `isDirExist`.

Reply:
If we're keeping this function, better to return an error and rename to `isFileAccessible` so we don't exit the program.